apps/src/music/utils/Patterns.ts

```diff
@@ -0,0 +1,63 @@
+import {
+  PatternEventValue,
+  PatternTickEvent
+} from '../player/interfaces/PatternEvent';
+import MusicLibrary from '../player/MusicLibrary';
+
+// This file contains a helper function for patterns, and is used by the
+// block's custom field.
+
+// A single event from a pattern to be rendered in a graph.
+export interface PatternGraphEvent {
```

Comment:
Can we share this interface with `Chords.ts`?

Reply:
Oh good idea.  I can do in a follow-up, since I'm still touching all of these controls as I keep them consistent through some refinements.